test/boost/sstable_test.cc

```diff
@@ -64,17 +64,25 @@ SEASTAR_TEST_CASE(composite_index) {
 }
 
 template<typename Func>
-inline auto
+inline future<>
 test_using_reusable_sst(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
-    return test_env::do_with([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
-        return env.reusable_sst(std::move(s), std::move(dir), generation_from_value(gen)).then([&env, func = std::move(func)] (sstable_ptr sst) mutable {
-            return func(env, std::move(sst));
-        });
+    return test_env::do_with_async([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
+        auto sst = env.reusable_sst(std::move(s), std::move(dir), generation_from_value(gen)).get();
+        futurize_invoke(func, env, std::move(sst)).get();
+    });
+}
+
+template<typename T, typename Func>
+inline future<T>
+test_using_reusable_sst_returning(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
```

Comment:
same here

Reply:
Although, you can deduce T. Declare func as `std::invocable<test_env&, shared_sstable> auto&& func>` and compute T using std::invoke_result_t.